plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -70,15 +70,18 @@ public void triggerScan(IWorkbenchWindow window) {
         IStructuredSelection structured = (IStructuredSelection) service.getSelection();
 
         Object firstElement = structured.getFirstElement();
-
+        IProject project = null;
+        if (firstElement instanceof JavaProject) {
+          project = ((JavaProject) firstElement).getProject();
+        }
+        
         if (firstElement instanceof IProject) {
-          IProject project = (IProject) firstElement;
-          runForProject(project.getName());
+          project = (IProject) firstElement;
         }
```

Comment:
If `firstElement` is a `JavaProject`, both conditions are true and the first assignment gets overwritten.

Reply:
Maybe you wanted an `else if`?

Reply:
How? A JavaProject is not in the same type hierarchy as IProject.

Reply:
You're right, I was confused by `IJavaProject` and the similar `getProject()` function